extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientImpl.java

```diff
@@ -120,9 +120,11 @@ private Tokens emitGrantTokens(HttpResponse<Buffer> resp, boolean refresh) {
             JsonObject json = resp.bodyAsJsonObject();
             final String accessToken = json.getString(oidcConfig.grant.accessTokenProperty);
             final String refreshToken = json.getString(oidcConfig.grant.refreshTokenProperty);
+            final Object expiresInValue = json.getValue(oidcConfig.grant.expiresInProperty);
             Long accessTokenExpiresAt;
-            Long accessTokenExpiresIn = json.getLong(oidcConfig.grant.expiresInProperty);
-            if (accessTokenExpiresIn != null) {
+            if (expiresInValue != null) {
+                long accessTokenExpiresIn = expiresInValue instanceof Long ? (Long) expiresInValue
+                        : Long.parseLong(expiresInValue.toString());
```

Comment:
Perhaps a better solution is: 

```suggestion
                long accessTokenExpiresIn = expiresInValue instanceof Number ? ((Number) expiresInValue).longValue()
                        : Long.parseLong(expiresInValue.toString());
```

Reply:
Clicked Resolve by mistake